backend/dataall/base/loader.py

```diff
@@ -185,8 +191,13 @@ def _initialize_modules(modes: Set[ImportMode]):
         raise ImportError("Not all modules have been initialized. Check if your import modes are correct")
 
 
-def _initialize_module(module):
+def _get_module_name(module):
+    return module[len(_MODULE_PREFIX) + 1:].split(".")[0]  # gets only top level module name
+
+
+def _initialize_module(module: Type[ModuleInterface]):
     module()  # call a constructor for initialization
+    _LOADED_MODULES.add(module.name())
 
```

Comment:
In line 219-230: Why do we have the double `while to_add` block? Why do we need to `to_add` and `new_to_add` variables?

Reply:
I implemented level by level BFS. It is working this way:
1) On the first iteration of first `while to_add` all active modules in the config file will be loaded.
2) If a module has a dependencies we will add them to `new_to_add` to load later (if they have not been loaded)
3) We switch `to_add` to `new_to_add` and repeat until all submodules are loaded